ocaml/xapi/vgpuops.ml

```diff
@@ -65,6 +65,7 @@ let allocate_vgpu_to_gpu ?(dry_run=false) ?(pre_allocate_list=[]) ~__context vm
   (* Make a asso list as follows
    * [(p1,capacity1);(p2,capacity2)...] *)
   let pgpu_capacity_assoc = List.intersect compatible_pgpus available_pgpus
+                            |> List.filter (fun pgpu -> Xapi_gpumon.Nvidia.vgpu_pgpu_are_compatible ~__context ~pgpu ~vgpu:vgpu.vgpu_ref) (* Filter all compatible pGPUs *)
```

Comment:
I believe we now rely on this function to be called by `Xapi_vm_helpers.assert_gpus_available` with `dry_run=true`, and when this gpumon check returns says that the GPU is incompatible, then an error is raised in L84.

Reply:
So did we somehow lose the compat check completely?